python/lsst/ap/pipe/metrics.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of ap_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Metrics for ap_pipe tasks.
+"""
+
+__all__ = [
+    "ApFakesCompletenessMetricTask", "ApFakesCompletenessMetricConfig",
+]
+
+import astropy.units as u
+import numpy as np
+
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import Struct
+import lsst.pipe.base.connectionTypes as connTypes
+from lsst.pipe.tasks.insertFakes import InsertFakesConfig
+from lsst.verify import Measurement, Name
+from lsst.verify.tasks import MetricTask, MetricComputationError
+
+
+class ApFakesCompletenessMetricConnections(
+        MetricTask.ConfigClass.ConnectionsClass,
+        dimensions={"instrument", "visit", "detector", "band"},
```

Comment:
I think the `band` is redundant here (it's implied by `visit`). Do you need to declare it to get the data ID extraction in `runQuantum` to work?